flink-core/src/main/java/org/apache/flink/api/common/time/Time.java

```diff
@@ -81,7 +82,24 @@ public String toString() {
 		return toMilliseconds() + " ms";
 	}
 
-	// ------------------------------------------------------------------------
+	@Override
```

Comment:
The problem is that `RestartStrategies` implement `equals/hashcode` methods that compare theirs `Time` fields. Because the `Time` class didn't have those methods their implementation was wrong.

I found out about it during testing, don't know if it affects any production code.

Reply:
👌